connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -0,0 +1,368 @@
+         * @return this builder to allow methods to be chained; never null
+         */
+        public NewTopicBuilder config(Map<String, Object> configs) {
+            if (configs != null) {
+                for (Map.Entry<String, Object> entry : configs.entrySet()) {
+                    Object value = entry.getValue();
+                    this.configs.put(entry.getKey(), value != null ? value.toString() : null);
+                }
+            } else {
+                this.configs.clear();
+            }
+            return this;
+        }
+
+        /**
+         * Build the {@link NewTopic} representation.
+         *
+         * @return the topic description; never null
+         */
+        public NewTopic build() {
+            return new NewTopic(name, numPartitions, replicationFactor).configs(configs);
+        }
+    }
+
+    /**
+     * Obtain a {@link NewTopicBuilder builder} to define a {@link NewTopic}.
+     *
+     * @param topicName the name of the topic
+     * @return the {@link NewTopic} description of the topic; never null
+     */
+    public static NewTopicBuilder defineTopic(String topicName) {
+        return new NewTopicBuilder(topicName);
+    }
+
+    private static final Logger log = LoggerFactory.getLogger(TopicAdmin.class);
+    private final Map<String, Object> adminConfig;
+    private final AdminClient admin;
+
+    /**
+     * Create a new topic admin component with the given configuration.
+     *
+     * @param adminConfig the configuration for the {@link AdminClient}
+     */
+    public TopicAdmin(Map<String, Object> adminConfig) {
+        this(adminConfig, AdminClient.create(adminConfig));
+    }
+
+    // visible for testing
+    TopicAdmin(Map<String, Object> adminConfig, AdminClient adminClient) {
+        this.admin = adminClient;
+        this.adminConfig = adminConfig != null ? adminConfig : Collections.<String, Object>emptyMap();
+    }
+
+    /**
+     * Filter the supplied list of topics to determine which of them already exist.
+     *
+     * @param topicNames the names of the topics
+     * @return the set of names that refer to already-existing topics; null only if the broker cannot perform this operation
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    private Set<String> existingTopics(Collection<String> topicNames) throws ExecutionException, InterruptedException {
+        if (topicNames == null || topicNames.isEmpty()) return Collections.emptySet();
+
+        Set<String> existingTopicNames = new HashSet<>();
+        try {
+            ListTopicsResults listingResults = admin.listTopics(new ListTopicsOptions().listInternal(true));
+            Map<String, TopicListing> topicListingsByName = listingResults.namesToDescriptions().get();
+            for (String topicName : topicNames) {
+                TopicListing listing = topicListingsByName.get(topicName);
+                if (listing != null) {
```

Comment:
This shouldn't be possible, right? It wouldn't make much sense to put a topic in the result if it didn't have a corresponding `TopicListing`.